File: src/Budgetify.Application/Accounts/UpdateAccount/AccountUpdateData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Budgetify.Domain.Accounts;

namespace Budgetify.Application.Accounts.UpdateAccount
{
    public sealed record AccountUpdateData(AccountId AccountId, AccountName Name, Money.DB.Money Balance)
    {
    }
}

File: src/client-app/src/models/requests/accountUpdateData.ts
import {Account} from "../accounts/account";

export interface AccountUpdateData {
    accountId: string
    name: string
    balance: number
}

export class AccountUpdateData implements AccountUpdateData {
    constructor(account: Account) {
        this.name = account.name
        this.balance = account.balance.amount
        this.accountId = account.accountId
    }
}